jetty-io/src/main/java/org/eclipse/jetty/io/ManagedSelector.java

```diff
@@ -115,19 +125,35 @@ protected void doStart() throws Exception
         // The normal strategy obtains the produced task, schedules
         // a new thread to produce more, runs the task and then exits.
         _selectorManager.execute(_strategy::produce);
+    }
 
-        // Set started only if we really are started
-        Start start = new Start();
-        submit(start);
-        start._started.await();
+    protected void onSelectFailed(Throwable cause) throws Exception
+    {
+        LOG.info("Restarting selector: " + toString(), cause);
+        startSelector();
+    }
+
+    private void notifySelectFailed(Throwable cause)
```

Comment:
I'm not sure it is worthwhile having this method, nor allowing `onSelectFailed` to throw.
Recovery from a selector failure is probably not achievable... recovery from an exception thrown by the recovery code on a selector failure certainly will never be recoverable.

Just don't have any declared exceptions and if they throw runtime, then we just break

Reply:
Having the `onSelectFailed()` works well enough.
Attempting to have an Exception flow out to somewhere reasonable (the Connector? the Server?) when the fault occurs deep within a ThreadPool > ExecutionStrategy > Connector > SelectorManager > ManagedSelector > Action is just awkward.

The updated testcase shows a possible way to use `onSelectFailed()` to restart the Server.

Reply:
I'm not sure you understood what I was saying.
`notifySelectFailed` is only called in one place, so inline it.  Also get rid of the try{}catch(){}  because we just cannot handle failures of listeners for failures.

Reply:
I've gone ahead and removed this notifySelectFailed() method.